src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java

```diff
@@ -81,57 +84,80 @@ public Vulnerability createVulnerability(Vulnerability vulnerability, boolean co
+                differ.applyIfChanged("cvssV2BaseScore", Vulnerability::getCvssV2BaseScore, existingVulnerability::setCvssV2BaseScore);
+                differ.applyIfChanged("cvssV2ImpactSubScore", Vulnerability::getCvssV2ImpactSubScore, existingVulnerability::setCvssV2ImpactSubScore);
+                differ.applyIfChanged("cvssV2ExploitabilitySubScore", Vulnerability::getCvssV2ExploitabilitySubScore, existingVulnerability::setCvssV2ExploitabilitySubScore);
+                differ.applyIfChanged("cvssV3Vector", Vulnerability::getCvssV3Vector, existingVulnerability::setCvssV3Vector);
+                differ.applyIfChanged("cvssV3BaseScore", Vulnerability::getCvssV3BaseScore, existingVulnerability::setCvssV3BaseScore);
+                differ.applyIfChanged("cvssV3ImpactSubScore", Vulnerability::getCvssV3ImpactSubScore, existingVulnerability::setCvssV3ImpactSubScore);
+                differ.applyIfChanged("cvssV3ExploitabilitySubScore", Vulnerability::getCvssV3ExploitabilitySubScore, existingVulnerability::setCvssV3ExploitabilitySubScore);
+                differ.applyIfChanged("owaspRRLikelihoodScore", Vulnerability::getOwaspRRLikelihoodScore, existingVulnerability::setOwaspRRLikelihoodScore);
+                differ.applyIfChanged("owaspRRBusinessImpactScore", Vulnerability::getOwaspRRBusinessImpactScore, existingVulnerability::setOwaspRRBusinessImpactScore);
+                differ.applyIfChanged("owaspRRTechnicalImpactScore", Vulnerability::getOwaspRRTechnicalImpactScore, existingVulnerability::setOwaspRRTechnicalImpactScore);
+                differ.applyIfChanged("owaspRRVector", Vulnerability::getOwaspRRVector, existingVulnerability::setOwaspRRVector);
+                differ.applyIfChanged("cwes", Vulnerability::getCwes, existingVulnerability::setCwes);
+                if (transientVulnerability.getVulnerableSoftware()!= null) {
+                    differ.applyIfChanged("vulnerableSoftware", Vulnerability::getVulnerableSoftware, existingVulnerability::setVulnerableSoftware);
             }
-            final Vulnerability result = persist(vulnerability);
+                // Persist changes only if there are differences
+                if (!differ.getDiffs().isEmpty()) {
+                    final Vulnerability result = persist(existingVulnerability);
             Event.dispatch(new IndexEvent(IndexEvent.Action.UPDATE, result));
             commitSearchIndex(commitIndex, Vulnerability.class);
             return result;
         }
-        return null;
+                // If no changes were made, return the existing vulnerability without modification
+                return existingVulnerability;
+                }
+            else {
+                // Handle cases where no existing vulnerability is found if needed (e.g., log an error)
+                LOGGER.warn("No existing vulnerability found for update operation.");
+                return new Vulnerability(); // return a new object or handle according to requirements
```

Comment:
Why not return `null`, or throw a `NoSuchElement` exception instead? 

Returning an empty object feels a bit ambiguous. Either this is an expected case and we return a corresponding result, or it is not and we `throw`.